terraform/modules/ci/builder_common/codebuild.tf

```diff
@@ -0,0 +1,81 @@
+locals {
+  github_event_patterns = ["PULL_REQUEST_CREATED", "PULL_REQUEST_UPDATED", "PULL_REQUEST_REOPENED"]
+}
+
+resource "aws_codebuild_project" "build" {
+  name          = var.project_name
+  description   = "For ${var.project_name}"
+  build_timeout = var.timeout_in_minutes
+
+  service_role = aws_iam_role.build.arn
+
+  vpc_config {
+    security_group_ids = var.agent_security_group_ids
+    subnets            = var.vpc_config.private_subnet_ids
+    vpc_id             = var.vpc_config.vpc_id
+  }
+
+  cache {
+    type  = "LOCAL"
+    modes = var.docker_required ? ["LOCAL_DOCKER_LAYER_CACHE", "LOCAL_SOURCE_CACHE"] : ["LOCAL_SOURCE_CACHE"]
+  }
+
+  environment {
+    compute_type                = "BUILD_GENERAL1_SMALL"
```

Comment:
Is this something we want to move into a var or we want a one size for everything?

Reply:
Good suggestion. Variable created for compute type.